eclair-core/src/test/scala/fr/acinq/eclair/db/PendingCommandsDbSpec.scala

```diff
@@ -0,0 +1,145 @@
+        new SqlitePendingCommandsDb(sqlite.connection)
+        new SqlitePendingCommandsDb(sqlite.connection)
+      case pg: TestPgDatabases =>
+        new PgPendingCommandsDb()(pg.datasource, pg.lock)
+        new PgPendingCommandsDb()(pg.datasource, pg.lock)
+    }
+  }
+
+  test("add/remove/list messages") {
+    forAllDbs { dbs =>
+      val db = dbs.pendingCommands
+
+      val channelId1 = randomBytes32()
+      val channelId2 = randomBytes32()
+      val msg0 = CMD_FULFILL_HTLC(0, randomBytes32())
+      val msg1 = CMD_FULFILL_HTLC(1, randomBytes32())
+      val msg2 = CMD_FAIL_HTLC(2, Left(randomBytes32()))
+      val msg3 = CMD_FAIL_HTLC(3, Left(randomBytes32()))
+      val msg4 = CMD_FAIL_MALFORMED_HTLC(4, randomBytes32(), FailureMessageCodecs.BADONION)
+
+      assert(db.listSettlementCommands(channelId1).toSet === Set.empty)
+      db.addSettlementCommand(channelId1, msg0)
+      db.addSettlementCommand(channelId1, msg0) // duplicate
+      db.addSettlementCommand(channelId1, msg1)
+      db.addSettlementCommand(channelId1, msg2)
+      db.addSettlementCommand(channelId1, msg3)
+      db.addSettlementCommand(channelId1, msg4)
+      db.addSettlementCommand(channelId2, msg0) // same messages but for different channel
+      db.addSettlementCommand(channelId2, msg1)
+      assert(db.listSettlementCommands(channelId1).toSet === Set(msg0, msg1, msg2, msg3, msg4))
+      assert(db.listSettlementCommands(channelId2).toSet === Set(msg0, msg1))
+      assert(db.listSettlementCommands().toSet === Set((channelId1, msg0), (channelId1, msg1), (channelId1, msg2), (channelId1, msg3), (channelId1, msg4), (channelId2, msg0), (channelId2, msg1)))
+      db.removeSettlementCommand(channelId1, msg1.id)
+      assert(db.listSettlementCommands().toSet === Set((channelId1, msg0), (channelId1, msg2), (channelId1, msg3), (channelId1, msg4), (channelId2, msg0), (channelId2, msg1)))
+    }
+  }
+
+  test("migrate database v1->v2") {
+    forAllDbs {
+      case dbs: TestPgDatabases =>
+        migrationCheck(
+          dbs = dbs,
+          initializeTables = connection => {
+            using(connection.createStatement()) { statement =>
+              statement.executeUpdate("CREATE TABLE pending_relay (channel_id TEXT NOT NULL, htlc_id BIGINT NOT NULL, data BYTEA NOT NULL, PRIMARY KEY(channel_id, htlc_id))")
+              setVersion(statement, "pending_relay", 1)
+            }
+            testCases.foreach { testCase =>
+              using(connection.prepareStatement("INSERT INTO pending_relay VALUES (?, ?, ?) ON CONFLICT DO NOTHING")) { statement =>
+                statement.setString(1, testCase.channelId.toHex)
+                statement.setLong(2, testCase.cmd.id)
+                statement.setBytes(3, cmdCodec.encode(testCase.cmd).require.toByteArray)
+                statement.executeUpdate()
+              }
+            }
+          },
+          dbName = "pending_relay",
+          targetVersion = 2,
+          postCheck = _ =>
```

Comment:
Do we have a way of checking the table name in the `postCheck`?

Reply:
We could by calling metadata tables, or even a `count` on the new table, but it feels a bit overkill, and calling `pendingCommands.listSettlementCommands()` does that indirectly.